engine/schema/src/org/apache/cloudstack/storage/datastore/db/PrimaryDataStoreDaoImpl.java

```diff
@@ -409,15 +460,13 @@ public StoragePoolVO persist(StoragePoolVO pool, Map<String, String> details) {
             sc.and(sc.entity().getScope(), Op.EQ, ScopeType.ZONE);
             return sc.list();
         } else {
-            Map<String, String> details = tagsToDetails(tags);
-
-            StringBuilder sql = new StringBuilder(ZoneWideDetailsSqlPrefix);
+            StringBuilder sql = new StringBuilder(ZoneWideTagsSqlPrefix);
```

Comment:
Thanks for pointing this out, I had missed it out. Created methods `getSqlPreparedStatement` and `searchStoragePoolsPreparedStatement` which are called from many methods and allow storage pool retrieval